scripts/common.py

```diff
@@ -712,3 +715,91 @@ def __enter__(self):
 
     def __exit__(self, etype, value, traceback):
         os.chdir(self.saved_path)
+
+
+@contextlib.contextmanager
+def managed_process(args, shell=False, **kwargs):
+    """Context manager for starting and stopping a process gracefully.
+
+    Args:
+        args: list(*). A sequence of program arguments, where the program to
+            execute is the first item. All items are converted to strings first.
+        shell: bool. Whether the command should be run inside of its own shell.
+        **kwargs: dict(str: *). Same kwargs as `subprocess.Popen`.
+
+    Yields:
+        psutil.Process. The process managed by the context manager.
+    """
+    if PSUTIL_DIR not in sys.path:
+        sys.path.insert(1, PSUTIL_DIR)
+    import psutil
+
+    str_args = (python_utils.UNICODE(arg).strip() for arg in args)
+    truthy_args = (arg for arg in str_args if arg)
+    popen_args = ' '.join(truthy_args) if shell else list(truthy_args)
+
+    root_proc = psutil.Popen(popen_args, shell=shell, **kwargs)
+
+    try:
+        yield root_proc
+    finally:
+        if root_proc.is_running():
+            procs = root_proc.children(recursive=True) + [root_proc]
+            for proc in procs:
+                proc.terminate()
+
+            _, procs_still_running = psutil.wait_procs(procs, timeout=10)
```

Comment:
Should we offer a bit more time for timeout? I would even say that second should be fine?